Source/CarthageKit/InputFilesInferrer.swift

```diff
@@ -0,0 +1,238 @@
+        return directory.appendingPathComponent(platform.relativePath, isDirectory: true)
+    }
+    
+    /// Maps Xcode's `FRAMEWORK_SEARCH_PATHS` string to an array or file URLs as well as resolves recursive directories.
+    ///
+    /// - Parameter rawFrameworkSearchPaths: Value of `FRAMEWORK_SEARCH_PATHS`
+    /// - Returns: Array of corresponding file URLs.
+    public static func frameworkSearchPaths(from rawFrameworkSearchPaths: String) -> [URL] {
+        // During expansion of the recursive path we don't want to enumarate over a directories that are known
+        // to be used as resources / other xcode-specific files that do not contain any frameworks.
+        let ignoredDirectoryExtensions: Set<String> = [
+            "app",
+            "dSYM",
+            "docset",
+            "framework",
+            "git",
+            "lproj",
+            "playground",
+            "xcassets",
+            "scnassets",
+            "xcstickers",
+            "xcbaseline",
+            "xcdatamodel",
+            "xcmappingmodel",
+            "xcodeproj",
+            "xctemplate",
+            "xctest",
+            "xcworkspace"
+        ]
+        
+        // We can not split by ' ' or by '\n' since it will give us invalid results because of the escaped spaces.
+        // To handle this we're replacing escaped spaces by ':' which seems to be the only invalid symbol on macOS,
+        // making conversion and then reverting replacement.
+        let escapingSymbol = ":"
+        return rawFrameworkSearchPaths
+            .replacingOccurrences(of: "\\ ", with: escapingSymbol)
+            .split(separator: " ")
+            .map { $0.replacingOccurrences(of: escapingSymbol, with: " ") }
+            .flatMap { path -> [URL] in
+                // For recursive paths Xcode adds a "**" suffix. i.e. /search/path turns into a /search/path/**
+                // We need to collect all the nested paths to act like an Xcode.
+                let recursiveSymbol = "**"
+                
+                if path.hasSuffix(recursiveSymbol) {
+                    let normalizedURL = URL(fileURLWithPath: String(path.dropLast(recursiveSymbol.count)), isDirectory: true)
+                    return FileManager.default.allDirectories(at: normalizedURL, ignoringExtensions: ignoredDirectoryExtensions)
+                } else {
+                    return [URL(fileURLWithPath: path, isDirectory: true)]
+                }
+            }
+            .map { $0.standardizedFileURL }
+    }
+}
+
+/// Invokes otool -L for a given executable URL.
+///
+/// - Parameter executableURL: URL to a valid executable.
+/// - Returns: Array of the Shared Library ID that are linked against given executable (`Alamofire`, `Realm`, etc).
+/// System libraries and dylibs are omited.
+internal func linkedFrameworks(for executable: URL) -> Result<[String], CarthageError> {
+    return Task("/usr/bin/xcrun", arguments: ["otool", "-L", executable.path.spm_shellEscaped()])
```

Comment:
this line doesn't work if the path contains spaces

Reply:
A common way to avoid the process of escaping characters is to execute: /bin/sh -c "<full command>". 
In this case:

`Task("/bin/sh", arguments: ["-c", "/usr/bin/xcrun otool -L \"\(executable.path)\""]`

Reply:
Fixed. Thanks for your suggestion, @werner77!